src/app-layout/visual-refresh/universal-toolbar.scss

```diff
@@ -37,20 +37,19 @@ section.universal-toolbar {
   > .toolbar-container {
     block-size: 48px;
     align-items: center;
-    display: inline-flex;
+    justify-content: space-between;
+    display: flex;
     column-gap: awsui.$space-static-xs;
-    inline-size: 100%;
```

Comment:
To only have drawers overflow when there isn't enough space, what about something like this:
```
    display: grid;
    column-gap: awsui.$space-static-s;
    inline-size: 100%;
    grid-template-columns: auto minmax(min(140px, 100%), 1fr);
    &:has(.universal-toolbar-nav) {
      grid-template-columns: min-content auto minmax(140px, 1fr);
    }
    grid-template-rows: var(#{custom-props.$toolbarHeight});
```

Reply:
Interesting. Let's go for the grid option, it gives us more convenient control how we collapse these items

Reply:
I'll add it to my next PR